lib/shopify_app/controller_concerns/token_exchange.rb

```
@@ -46,64 +51,13 @@ def current_shopify_domain
-        session_token: session_token,
-        requested_token_type: ShopifyAPI::Auth::TokenExchange::RequestedTokenType::OFFLINE_ACCESS_TOKEN,
-      )
-
-      if session && online_token_configured?
-        ShopifyApp::Logger.info("Performing Token Exchange for [#{domain}] - (Online)")
-        session = exchange_token(
-          shop: domain, # TODO: use jwt_shopify_domain ?
-          session_token: session_token,
-          requested_token_type: ShopifyAPI::Auth::TokenExchange::RequestedTokenType::ONLINE_ACCESS_TOKEN,
-        )
-      end
-
-      ShopifyApp.configuration.post_authenticate_tasks.perform(session)
-    end
-
-    def exchange_token(shop:, session_token:, requested_token_type:)
-      if session_token.blank?
-        # respond_to_invalid_session_token
-        return
-      end
-
-      begin
-        session = ShopifyAPI::Auth::TokenExchange.exchange_token(
-          shop: shop,
-          session_token: session_token,
-          requested_token_type: requested_token_type,
-        )
-      rescue ShopifyAPI::Errors::InvalidJwtTokenError
-        # respond_to_invalid_session_token
-        return
-      rescue ShopifyAPI::Errors::HttpResponseError => error
-        ShopifyApp::Logger.error(
-          "A #{error.code} error (#{error.class}) occurred during the token exchange. Response: #{error.response.body}",
-        )
-        raise
-      rescue => error
-        ShopifyApp::Logger.error("An error occurred during the token exchange: #{error.message}")
-        raise
-      end
-
-      if session
-        begin
-          ShopifyApp::SessionRepository.store_session(session)
-        rescue ActiveRecord::RecordNotUnique
-          ShopifyApp::Logger.debug("Session not stored due to concurrent token exchange calls")
-        end
-      end
-
-      session
+      ShopifyApp::Auth::TokenExchange.perform(session_token)
+    rescue ShopifyAPI::Errors::InvalidJwtTokenError
+      # respond_to_invalid_session_token
```

This is silently ignoring this error - is that something we want to do?